src/fparser/two/utils.py

```diff
@@ -702,8 +702,42 @@ def match(separator, subcls, string):
         if not splitted:
             # There should be at least one entry.
             return None
-        lst = [subcls(repmap(entry.strip())) for entry in splitted]
-        return separator, tuple(lst)
+
```

Comment:
I can't see anything obviously wrong with this although it does seem a bit complicated.
For the case when separator is white space wouldn't a regex to remove white space after opening and before closing parentheses(/other bracket symbols) be simpler? i.e. fix the content of 'splitted' rather than trying to retrospectively match on bits of it?

Reply:
I've reworked the solution after discussion.